tools/base/utils.py

```diff
@@ -74,14 +79,19 @@ def buffered(
         stderr.extend(mangle(_stderr.read().strip().split("\n")))
 
 
-def extract(tarball: str, path: str) -> str:
-    with tarfile.open(tarball) as tarfiles:
-        tarfiles.extractall(path=path)
-        return path
+def extract(path: Union[pathlib.Path, str], *tarballs: Union[pathlib.Path,
+                                                             str]) -> Union[pathlib.Path, str]:
+    if not tarballs:
+        raise ExtractError(f"No tarballs specified for extraction to {path}")
+    openers = nested(*tuple(tarfile.open(tarball) for tarball in tarballs))
+    with openers as tarfiles:
+        for _tarfile in tarfiles:
```

Comment:
Nit: dislike `_` prefix for locals, it confuses with private object variables.

Reply:
updated